### ISSUE TITLE
tauri: failed to parse JSON-RPC request

### ISSUE THREAD

**Issue by Gussy** [oscartbeaumont/rspc#77]
https://github.com/tauri-apps/tauri/pull/5492 which was recently merged into the `dev` branch has broken the tauri integration. All RPC calls fail with the following error message:
```
2022-10-30T11:37:47.115378Z ERROR rspc::integrations::tauri: failed to parse JSON-RPC request: invalid type: string "{\"id\":\"b8h6sjydy9l\",\"method\":\"query\",\"params\":{\"path\":\"version\"}}", expected struct Request at line 1 column 83
```

I tried some hacks to fix the issue on the rust and typescript sides of rspc but wasn't successful. Happy to help PR a fix with a nudge in the right direction.

**Versions:**
tauri: `dev` branch
rspc: v1.0.2 (also fails with `main`)

**Comment by RadoslavK** [comment]
Might be why the app stopped working after upgrading tauri to v1.2.0.

**Comment by oscartbeaumont** [comment]
This should be fixed as part of #84.

Right now upgrading rspc to the version in Git is a little bit challenging as the `main` branch is using a completly refactored Typescript client and a slightly different format for frontend/backend communication all of which isn't documented yet. I am going to work on getting a release out in December once I am back from travelling so hopefully it's possible for you to wait out until then. If not, feel free to let me know in the Discord and I can give you a hand upgrading.
